tycho-source-plugin/src/main/java/org/eclipse/tycho/source/OsgiSourceMojo.java

```diff
@@ -61,10 +61,10 @@
 /**
  * Goal to create a JAR-package containing all the source files of a osgi project.
  */
-@Mojo(name = "plugin-source", defaultPhase = LifecyclePhase.PREPARE_PACKAGE, threadSafe = true)
```

Comment:
Do you think it is in general a reasonable pattern to use a constant as mojo-name (if available)?

Reply:
I think it doesn't hurt.

Reply:
Do you think it would be a good pattern to be applied generally to have a `public static final String GOAL` field for each Mojo class that could be referenced wherever the goal id is required? It would avoid using 'magic' string literals in the code base.

Reply:
I think it does not makes much of a difference if it is in a `GOAL="..."` or in a `name = "..."` unless we reference it somewhere else (in which case it might be more useful in `TychoConstants` or a new `TychoMojos` interface)

Reply:
Sure it only makes a difference when the goal-id is used somewhere else.
The good think about such constants is that in Eclipse one can simply check the call-hierarchy to see where the goal is used and does not have to search for the String and sort out all false positive results.

I'm not sure where such constant would be placed best. My first thought was the mojo because it's a property of it.
But I'm not planning to apply this to the entire code base at the moment. So there is some time to keep this in the back of our minds.

Reply:
The mojo ids are at least used in the compontents.xml and we cant reference the constants there.